x-pack/legacy/plugins/siem/server/lib/detection_engine/rules/read_rules.test.ts

```diff
@@ -21,6 +37,49 @@ describe('read_rules', () => {
       });
       expect(rule).toEqual(getResult());
     });
+    test('should return null if saved object found by alerts client given id is not alert type', async () => {
+      const alertsClient = alertsClientMock.create();
+      const { alertTypeId, ...rest } = getResult();
+      // @ts-ignore
+      alertsClient.get.mockImplementation(() => rest);
```

Comment:
Typescript is your biggest best friends of best friends here. Never ts-ignore your friends. ;-)

This looks like what you really intended for was this:

```ts
      const { alertTypeId, ...rest } = getResult();
      alertsClient.get.mockResolvedValue((rest as unknown) as RuleAlertType);
```

or another way that is still type safe would be just a mutatious delete like so:

```ts
      const alertsClient = alertsClientMock.create();
      const result = getResult();
      delete result.alertTypeId;
      alertsClient.get.mockResolvedValue(result);
```

Then you don't have to worry about doing odd casting since delete is a side effect and get no `unknown`.

What's tricky is that this test is incredibly easy to have false positives and by deleting this code the test still passes, so buyer beware.

Reply:
I will revisit this one. I definitely don't want false positives.